File: backend/src/main/java/de/neoventus/rest/dto/ReservationDto.java
package de.neoventus.rest.dto;

import org.hibernate.validator.constraints.NotEmpty;

import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.Date;

/**
 * @author Tim Heidelbach
 * @author Dennis Thanner
 */
public class ReservationDto implements Serializable {

	private String id;

	@NotNull
	@NotEmpty
	private String[] desk;

	@NotNull
	private Date time;

	private Integer duration;

	private String reservedBy;


	private String reservationName;

	// Getter and setter

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getReservedBy() {
		return reservedBy;
	}

	public void setReservedBy(String reservedBy) {
		this.reservedBy = reservedBy;
	}

	public String[] getDesk() {
		return desk;
	}

	public void setDesk(String[] desk) {
		this.desk = desk;
	}

	public Date getTime() {
		return time;
	}

	public void setTime(Date time) {
		this.time = time;
	}

	public Integer getDuration() {
		return duration;
	}

	public void setDuration(Integer duration) {
		this.duration = duration;
	}

	public String getReservationName() {
		return reservationName;
	}

	public void setReservationName(String reservationName) {
		this.reservationName = reservationName;
	}
}

File: frontend/src/model/reservation-dto.ts
/**
 * @author Markus Knauer
 * @version 0.0.1
 */
export class ReservationDto {

  private id: string;

  private desk: Array<string>;

  private time: Date;

  private duration: number;

  private reservedBy: string;

  private reservationName: string;

  // Getter and setter

  get Id() {
    return this.id;
  }

  set Id(id: string) {
    this.id = id;
  }

  get ReservedBy() {
    return this.reservedBy;
  }

  set ReservedBy(reservedBy: string) {
    this.reservedBy = reservedBy;
  }

  get Desk() {
    return this.desk;
  }

  set Desk(desk: Array<string>) {
    this.desk = desk;
  }

  get Time() {
    return this.time;
  }

  set Time(time: Date) {
    this.time = time;
  }

  get Duration() {
    return this.duration;
  }

  set Duration(duration: number) {
    this.duration = duration;
  }

  set ReservationName(reservationName: string) {
    this.reservationName = reservationName;
  }

  get ReservationName() {
    return this.reservationName;
  }

}
